modules/components/src/AdvancedFacetView/LiveAdvancedFacetView.js

```diff
@@ -125,7 +132,15 @@ export default class LiveAdvancedFacetView extends React.Component {
   }
   onSqonFieldChange = ({ sqon }) => {
     const { onSqonChange = () => {} } = this.props;
-    this.setState({ sqon }, () => onSqonChange({ sqon }));
+    fetchAggregationData({
+      ...this.props,
+      extended: this.state.extended.filter(
+        e => e.type !== 'object' && e.type !== 'nested',
+      ),
+      sqon,
+    }).then(({ aggregations }) =>
+      this.setState({ sqon, aggregations }, () => onSqonChange({ sqon })),
```

Comment:
This means all the values that are irrelevant for a given `sqon` will be filtered out as it will not be included in the aggregations returned. This feels like a nice feature in my opinion but good to note.

Reply:
So turned out, there is an argument for this in the `aggregations` query, but it's not working.
Tracking this as a separate issue here: https://github.com/overture-stack/arranger/issues/168